Content.Server/GameObjects/EntitySystems/DisassembleOnActivateSystem.cs

```diff
@@ -17,16 +21,23 @@ public override void Initialize()
             SubscribeLocalEvent<DisassembleOnActivateComponent, ActivateInWorldMessage>(HandleActivateInWorld);
         }
 
+        public override void Shutdown()
+        {
+            base.Shutdown();
+
+            UnsubscribeLocalEvent<DisassembleOnActivateComponent, ActivateInWorldMessage>(HandleActivateInWorld);
+        }
+
         private async void HandleActivateInWorld(EntityUid uid, DisassembleOnActivateComponent component, ActivateInWorldMessage args)
         {
-            if (component.Prototype == null)
+            if (string.IsNullOrEmpty(component.Prototype))
                 return;
             if (!args.User.InRangeUnobstructed(args.Activated))
                 return;
 
             if (component.DoAfterTime > 0 && TryGet<DoAfterSystem>(out var doAfterSystem))
             {
-                var doAfterArgs = new DoAfterEventArgs(args.User, component.DoAfterTime)
+                var doAfterArgs = new DoAfterEventArgs(args.User, component.DoAfterTime, component.TokenSource.Token)
```

Comment:
Create a new token cancellation source and cancel the old one before passing this one here.